gear-programs/eth-events-common/src/lib-template.rs

```diff
@@ -0,0 +1,75 @@
+use cell::RefCell;
+use eth_events_common::{CheckedProofs, Error, Proofs, State};
+use ethereum_common::{beacon::BlockHeader as BeaconBlockHeader, tree_hash::TreeHash, H256};
+use sails_rs::prelude::*;
+
+pub struct EthereumEventClientProgram(RefCell<State>);
+
+#[sails_rs::program]
+impl EthereumEventClientProgram {
+    pub fn new(checkpoint_light_client_address: ActorId) -> Self {
+        Self(RefCell::new(State {
+            checkpoint_light_client_address,
+        }))
+    }
+
+    pub fn ethereum_event_client(&self) -> Service {
+        Service::new(&self.0)
+    }
+}
+
+#[derive(Clone, Debug, Decode, TypeInfo)]
+#[codec(crate = sails_rs::scale_codec)]
+#[scale_info(crate = sails_rs::scale_info)]
+pub struct BlockInclusionProof {
+    pub block: LightBeaconBlock,
+    pub headers: Vec<BeaconBlockHeader>,
+}
+
+#[derive(Clone, Debug, Decode, TypeInfo)]
+#[codec(crate = sails_rs::scale_codec)]
+#[scale_info(crate = sails_rs::scale_info)]
+pub struct EthToVaraEvent {
+    pub proof_block: BlockInclusionProof,
+    pub proof: Vec<Vec<u8>>,
+    pub transaction_index: u64,
+    pub receipt_rlp: Vec<u8>,
+}
+
+pub struct Service<'a> {
+    state: &'a RefCell<State>,
+}
+
+#[sails_rs::service]
+impl<'a> Service<'a> {
+    pub fn new(state: &'a RefCell<State>) -> Self {
+        Self { state }
+    }
+
+    pub fn checkpoint_light_client_address(&self) -> ActorId {
+        self.state.borrow().checkpoint_light_client_address
+    }
+
+    pub async fn check_proofs(&mut self, message: EthToVaraEvent) -> Result<CheckedProofs, Error> {
```

Comment:
It would be also nice to add a test to historical proxy that routes of both `checkpoint_light_client_address` & `check_proofs` are the same for `*-deneb`/ `*-electra`.

Just a test with simple `assert_eq` 's on `ActionIo::ROUTE`.